File: examples/sdk/typescript/01-quick-start.ts
/**
 * Example 01: Quick Start — 5 minutes to your first search
 *
 * Demonstrates: Simple API, auto-auth from ~/.kweaver/, BKN listing, semantic search.
 *
 * Run: npx tsx examples/sdk/01-quick-start.ts
 */
// Monorepo import — published users would use: import kweaver from "@kweaver-ai/kweaver-sdk/kweaver";
import kweaver from "../../../packages/typescript/src/kweaver.js";

async function main() {
  // 1. Configure — reads credentials from ~/.kweaver/ automatically
  kweaver.configure({ config: true });
  console.log("✓ Configured from ~/.kweaver/\n");

  // 2. List available knowledge networks
  const knList = await kweaver.bkns({ limit: 10 });
  console.log(`Found ${knList.length} knowledge network(s):`);
  for (const kn of knList) {
    const item = kn as { id?: string; name?: string };
    console.log(`  - ${item.name} (${item.id})`);
  }

  if (knList.length === 0) {
    console.log("\nNo BKNs found. Create one first.");
    return;
  }

  // 3. Pick the first BKN and do a semantic search
  const first = knList[0] as { id?: string; name?: string };
  const bknId = first.id!;
  console.log(`\nSearching in "${first.name}"...`);

  // "数据" means "data" in Chinese — change this to match your BKN's language
  const result = await kweaver.search("数据", { bknId, maxConcepts: 5 });
  console.log(`\nSearch results (${result.hits_total ?? 0} hits):`);
  for (const concept of result.concepts ?? []) {
    const c = concept as { concept_name?: string; intent_score?: number };
    console.log(`  - ${c.concept_name} (score: ${c.intent_score})`);
  }
}

main().catch(console.error);

File: examples/sdk/python/01-quick-start.py
"""Example 01: Quick Start — 5 minutes to your first search.

Demonstrates: module-level API, auto-auth from ``~/.kweaver/``, BKN listing,
semantic search.

Run: python examples/sdk/01-quick-start.py
"""

from __future__ import annotations

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[3] / "packages" / "python" / "src"))

import kweaver  # noqa: E402


def main() -> None:
    kweaver.configure(config=True)
    print("✓ Configured from ~/.kweaver/\n")

    kn_list = kweaver.bkns(limit=10)
    print(f"Found {len(kn_list)} knowledge network(s):")
    for kn in kn_list:
        print(f"  - {kn.name} ({kn.id})")

    if not kn_list:
        print("\nNo BKNs found. Create one first.")
        return

    first = kn_list[0]
    print(f'\nSearching in "{first.name}"...')

    # "数据" means "data" in Chinese — change this to match your BKN's language
    result = kweaver.search("数据", bkn_id=first.id, max_concepts=5)
    print(f"\nSearch results ({result.hits_total or 0} hits):")
    for c in result.concepts or []:
        print(f"  - {c.concept_name} (score: {c.intent_score})")


if __name__ == "__main__":
    main()
